src/ReplaySession.cc

```diff
@@ -297,20 +307,21 @@ Completion ReplaySession::enter_syscall(ReplayTask* t,
       // in writeable or shared memory, since in those cases it could be
       // overwritten by the tracee. It could even be dynamically generated and
       // not generated yet.
-      if (t->vm()->is_breakpoint_in_private_read_only_memory(
+      if (!syscall_bp_vm &&
+          t->vm()->is_breakpoint_in_private_read_only_memory(
               syscall_instruction) &&
           t->vm()->add_breakpoint(syscall_instruction, BKPT_INTERNAL)) {
```

Comment:
That would be a good sanity check, but I don't think there ever should be a situation where they don't? I'll try adding it as an assertion and see what happens.